DashAI/back/tasks/base_task.py

```diff
@@ -214,7 +132,24 @@ def validate_dataset_for_task(self, dataset: DatasetDict):
                 and len(dataset[split].outputs_columns) != outputs_cardinality
             ):
                 raise ValueError(
+                    f"Error in split {split} of dataset {dataset_name}. "
                     f"Output cardinality ({len(dataset[split].outputs_columns)})"
                     f" does not "
                     f"match task cardinality ({outputs_cardinality})"
                 )
+
+    @abstractmethod
+    def prepare_for_task(self, dataset: DatasetDict):
+        """Method to change the column types to suit the task
+
```

Comment:
Include a comment that indicate if the method will create a copy or with mutate the original object.